In [1]:
import pandas as pd

# Load JSON file
json_file_path = 'spotify.json'
data = pd.read_json(json_file_path)

# Convert JSON data to CSV
csv_file_path = 'output.csv'
data.to_csv(csv_file_path, index=False)


In [11]:
spotify = pd.read_csv("output.csv")
spotify["dateTime"] = pd.to_datetime(spotify["endTime"])
spotify

spotify.sort_values('dateTime', inplace=True)

# Calculate time difference between each row
spotify['time_difference'] = spotify['dateTime'].diff()

# Convert time difference to timedelta object
spotify['time_difference'] = pd.to_timedelta(spotify['time_difference'])

spotify


endTime                             artistName  \
0     2023-01-03 04:26                             Fujii Kaze   
1     2023-01-03 17:33  Música Piola Que No Estaba En Spotify   
2     2023-01-04 01:00  Música Piola Que No Estaba En Spotify   
3     2023-01-04 01:01                             Pete & Bas   
4     2023-01-04 01:01                                 Yoshii   
...                ...                                    ...   
3021  2023-04-20 19:26                                    YUI   
3022  2023-04-20 19:30                     MAN WITH A MISSION   
3023  2023-04-20 19:32                                    YUI   
3024  2023-04-20 19:35                          Palmy Chiller   
3025  2023-04-20 19:38                          almost monday   

                                              trackName  msPlayed  \
0                                        Shinunoga E-Wa      8815   
1     Seishun Buta Yarou wa Bunny Girl Senpai no Yum...     10726   
2     Seishun Buta Yarou wa Bunny Girl Senpai no Yum...     17595   
3                                          Mr Worldwide     32401   
4                                    Fukashigi No Carte       801   
...                                                 ...       ...   
3021                                               LIFE     31910   
3022                                               絆ノ奇跡    223320   
3023                                               LIFE    152440   
3024                             Why Do We Have to End?    179354   
3025                                            sunburn    173218   

                dateTime time_difference  
0    2023-01-03 04:26:00             NaT  
1    2023-01-03 17:33:00 0 days 13:07:00  
2    2023-01-04 01:00:00 0 days 07:27:00  
3    2023-01-04 01:01:00 0 days 00:01:00  
4    2023-01-04 01:01:00 0 days 00:00:00  
...                  ...             ...  
3021 2023-04-20 19:26:00 0 days 00:01:00  
3022 2023-04-20 19:30:00 0 days 00:04:00  
3023 2023-04-20 19:32:00 0 days 00:02:00  
3024 2023-04-20 19:35:00 0 days 00:03:00  
3025 2023-04-20 19:38:00 0 days 00:03:00  

[3026 rows x 6 columns]

In [3]:
# find largest time difference
maxValue = spotify["time_difference"].max()

# find the average time differnce
avgValue = spotify["time_difference"].mean()

# Print the largest value
print("The largest difference in time is", maxValue)

# Print the average time difference
print("The average difference in time is", avgValue)


The largest difference in time is 11 days 10:27:00
The average difference in time is 0 days 00:51:14.221487603


In [14]:
# Extracts month from datetime object
spotify['month'] = spotify['dateTime'].dt.month

# Find the month with the highest count
month = spotify['month'].value_counts().idxmax()

# Print the most common month
print("The most common month is:", month)

# Find the ammount of times that month appears in the collumn
count = (spotify["month"] == month).sum()

# Print the number of times the month appears
print("The month of", month, "appears", count, "times")

spotify['day'] = spotify['dateTime'].dt.weekday

day = spotify['day'].mode().values[0]

# int to day
dayNames = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
dayName = dayNames[day]

print("The most common day is", dayName)

The most common month is: 1
The month of 1 appears 1609 times
The most common day is Friday


The reason for this is simply that the most music was listened to in the month of January. There isn't any extenuating circumstances, it is most likely purely coincidence that the most songs were played in jan, amounting to a total of 1609 plays that month.Also the fact that the most songs were listened to on friday is also likely to be a coincidence. 

The sample of data is not super large and in order to prove some kind of corelation between the month and amount of listens, we would need more years of data to prove it. 

In [6]:
artist = spotify["artistName"].value_counts().idxmax()

print("The most played artist is", artist)


The most played artist is blink-182
